wukong/request.py

```diff
@@ -27,105 +31,135 @@ class SolrRequest(object):
     """
     Handle requests to SOLR and response from SOLR
     """
+
+    REFRESH_FREQUENCY = 2
+
     def __init__(self, solr_hosts, zookeeper_hosts=None, timeout=15):
         self.client = requests.Session()
         self.master_hosts = solr_hosts
-        self.current_hosts = solr_hosts
+        self.zookeeper_hosts = zookeeper_hosts
         self.servers = []
         self.timeout = timeout
-        if zookeeper_hosts is not None:
-            self.zookeeper = Zookeeper(zookeeper_hosts)
-        else:
-            self.zookeeper = None
-
-        self.last_error = None
-        # time to revert to old host list (in minutes) after an error
-        self.check_hosts = 5
-
-    def request(self, path, params, method, body=None):
+        self.current_hosts = self.master_hosts  # Backwards Compat
+        self._zookeeper = None
+        self._last_request = None
+        self.attempt_zookeeper_refresh()
+
+    @property
+    def zookeeper(self):
+        if self._zookeeper is None and self.zookeeper_hosts:
+            self._zookeeper = Zookeeper(self.zookeeper_hosts)
+        return self._zookeeper
+
+    def attempt_zookeeper_refresh(self):
+        zk = self.zookeeper
+        if zk:
+            logger.debug('Fetching solr from zookeeper')
+            try:
+                self.master_hosts = zk.get_active_hosts()
+                self.current_hosts = self.master_hosts  # backward compat
```

Comment:
We should make
```py
@property
def current_hosts(self):
    return self.master_hosts
```
and then we don't have to worry about keeping it updated

Reply:
🌮